src/otg/preprocess/finngen/study_index.py

```diff
@@ -0,0 +1,128 @@
+"""FinnGen ingestion: study index."""
+
+from __future__ import annotations
+
+import logging
+import sys
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+from urllib.request import urlopen
+
+import pyspark.sql.functions as f
+from pyspark.sql import SparkSession
+
+from otg.common.schemas import parse_spark_schema
+from otg.dataset.study_index import StudyIndex
+
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame
+    from pyspark.sql.types import StructType
+
+logging.basicConfig(level=logging.INFO, format="%(asctime)s %(message)s")
+
+
+@dataclass
+class FinnGenStudyIndex(StudyIndex):
+    """Study index dataset from FinnGen.
+
+    The following information is aggregated/extracted:
+
+    - Study ID in the special format (FINNGEN_R9_*)
+    - Trait name (for example, Amoebiasis)
+    - Number of cases and controls
+    - Link to the summary statistics location
+
+    Some fields are also populated as constants, such as study type and the initial sample size.
+    """
+
+    @classmethod
+    def get_schema(cls: type[FinnGenStudyIndex]) -> StructType:
```

Comment:
You don't need this function. You get it by inheritance from StudyIndex

Reply:
Thank you! This was an abstract method before, and now this is indeed not required. Fixed.